freqtrade/plugins/pairlist/RemotePairList.py

```diff
@@ -0,0 +1,166 @@
+        if self._bearer_token:
+            headers['Authorization'] = f'Bearer {self._bearer_token}'
+
+        info = "Pairlist"
+
+        try:
+            response = requests.get(self._pairlist_url, headers=headers,
+                                    timeout=self._read_timeout)
+            content_type = response.headers.get('content-type')
+            time_elapsed = response.elapsed.total_seconds()
+
+            if "application/json" in str(content_type):
+                jsonparse = response.json()
+                pairlist = jsonparse['pairs']
+                info = jsonparse.get('info', '')
+            else:
+                raise OperationalException('RemotePairList is not of type JSON abort ')
+
+            self._refresh_period = jsonparse.get('refresh_period', self._refresh_period)
+            self._pair_cache = TTLCache(maxsize=1, ttl=self._refresh_period)
+
+        except requests.exceptions.RequestException:
+            self.log_once(f'Was not able to fetch pairlist from:'
+                          f' {self._pairlist_url}', logger.info)
+
+            if self._keep_pairlist_on_failure:
+                pairlist = self._last_pairlist
+                self.log_once('Keeping last fetched pairlist', logger.info)
+            else:
+                pairlist = []
+
+            time_elapsed = 0
+
+        return pairlist, time_elapsed, info
+
+    def gen_pairlist(self, tickers: Tickers) -> List[str]:
+        """
+        Generate the pairlist
+        :param tickers: Tickers (from exchange.get_tickers). May be cached.
+        :return: List of pairs
+        """
+
+        pairlist = self._pair_cache.get('pairlist')
+        time_elapsed = 0.0
+
+        if pairlist:
+            # Item found - no refresh necessary
+            return pairlist.copy()
+        else:
+            if self._pairlist_url.startswith("file:///"):
+                filename = self._pairlist_url.split("file:///", 1)[1]
+                file_path = Path(filename)
+
+                if file_path.exists():
+                    with open(filename) as json_file:
+                        # Load the JSON data into a dictionary
+                        jsonparse = json.load(json_file)
+                        pairlist = jsonparse['pairs']
+                        info = jsonparse.get('info', '')
+                        self._refresh_period = jsonparse.get('refresh_period', self._refresh_period)
+                        self._pair_cache = TTLCache(maxsize=1, ttl=self._refresh_period)
+
+                else:
+                    raise ValueError(f"{self._pairlist_url} does not exist.")
+            else:
+                # Fetch Pairlist from Remote URL
+                pairlist, time_elapsed, info = self.fetch_pairlist()
+
+        pairlist = self.filter_pairlist(pairlist, tickers)
+        self._pair_cache['pairlist'] = pairlist.copy()
+
+        if time_elapsed != 0.0:
+            self.log_once(f'{info} Fetched in {time_elapsed} seconds.', logger.info)
```

Comment:
I'd be very careful with this.
we're efectively directly logging remote content.

the pairlist is different, as that's filtered first (so only actual valid pairs would be logged) - but here we could get anything (also malicious) - thinking back to log4J last year - that's not something i'd want to find bot users in ...

Reply:
We could check for a max-length of 256 chars, only alphanumeric and wrap it in a try: except: clause, would that be sufficient or would it have to be left out?

Reply:
what i mean is - what's the (expected) benefit of logging an additional info?

Limiting to something is fine - but once we need to do full input validation (or do whatever to be sure it's not problematic) i'd question if we "really" need it.

Reply:
it could return additional statistics, or return filtered pairs for example i used it for 

"info":"Fear and Greed:26, BTC:16723.52USD, BTC Dominance:39.14%, BTC price changed by +0.29% from yesterday to today. Updated:13:01UTC."

or return the sorting / filters used:
"info":" # Pairs: 40, Filtered: 14, Sort by marketcap (50) desc + rsi_14 (40) desc

its not per se needed but it could give information about the pairlist.

Reply:
keep in mind that it's information that's only shown in the logs - so you'll need to actively look for it to actually see this.

Reply:
its no must i will remove it from the code to be safe